torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1991,8 +1990,9 @@ def _maybe_evaluate_static_worker(
             # This is similar to the flavor where size oblivious omits
             # 0/1, it changes semantics but in a benign way.
             upper = min(2**48, vr.upper)
-            if oblivious_upper_bound_exclusive is not None:
-                upper = min(upper, oblivious_upper_bound_exclusive - 1)
+            # Excluding the very upper bound can be helpful
```

Comment:
maybe include an example where this previously would have failed? bonus points for a test

Reply:
This change subsumes https://github.com/pytorch/pytorch/pull/144471 , so the test from that one is good